kotlin-checks-test-sources/src/main/kotlin/checks/VarShouldBeValCheckSample.kt

```diff
@@ -0,0 +1,114 @@
+package checks
+
+import kotlin.math.max
+
+class VarShouldBeValCheckSample {
+    private var foo = 0 // Noncompliant {{Replace the keyword `var` with `val`.}}
+    private val bar = 0 // compliant
+    var notUsed = "not used" // compliant, not a local variable
+    private var y = "y" // Noncompliant
+        set(value){
+            field = value
+        }
+    private var abc = 0 // compliant
+
+    fun classProperties(): Unit {
+        this::abc.set(1)
+    }
+
+
+    class Address {
+        var street: String = "baker" // compliant
+        var number: Int = 221 // compliant
+    }
+
+    annotation class Fancy
+
+    fun assignmentOperators(): Unit {
+        @Fancy
+        var j = 0 // Noncompliant
+
+        var x = 0 // compliant
+        x = 1
+        var y = 0 // compliant
+        y += 1
+        var p = 0 // compliant
+        (p) += 1
+        var i = 0 // compliant
+        i -= 1
+        var z = 0 // compliant
+        z++
+        var w = 0 // compliant
+        w--
+        var k = 0 // compliant
+        ++k
+        var l = 0 // compliant
+        --l
+        var v = 0 // compliant
+        v *= 2
+        var u = 0 // compliant
+        u/= 2
+        var t = 0 // compliant
+        t %= 2
+        var (a, b) = Pair(0, 1) // Noncompliant
```

Comment:
We should probably not raise anything if one of the restructured vars is written to

Reply:
:+1: